client/comms/wsconn_test.go

```diff
@@ -194,14 +194,19 @@ func TestWsConn(t *testing.T) {
 	defer server.Shutdown(context.Background())
 
 	wg.Add(1)
+	serverReady := make(chan struct{}, 1)
 	go func() {
 		defer wg.Done()
+		serverReady <- struct{}{}
 		err := server.ListenAndServeTLS(certFile.Name(), keyFile.Name())
 		if err != nil {
 			fmt.Println(err)
 		}
 	}()
 
+	// wait for server to start listening before connecting
+	<-serverReady
+
```

Comment:
I thought about adding a sleep here as well, but this should suffice, right?  This ensures that the server listen goroutine has begun execution before connections are attempted, but the server actually starts listening a very brief moment later, so before connections are attempted, the server should already be listening.

Reply:
The catch is that the channel send happens before it begins listening.  I don't see any way around a sleep unless we abandon `ListenAndServeTLS` in favor of something like

```
listen()
serverReady <- struct{}{}
serve()
```

Reply:
What you've described is basically what `ListenAndServeTLS` does, so that's a preferred approach I think.